cmd/gardener-extension-acl/app/app.go

```diff
@@ -103,6 +105,12 @@ func (o *Options) run(ctx context.Context) error {
 		return fmt.Errorf("could not add controllers to manager: %s", err)
 	}
 
+	// TODO(Wieneo): Remove this once a couple extension versions included the migration code
+	// migration code: remove mutating webhook from cluster as it is not served by this controller anymore
+	if err := client.IgnoreNotFound(mgr.GetClient().Delete(ctx, &admissionregistrationv1.MutatingWebhookConfiguration{ObjectMeta: metav1.ObjectMeta{Name: ExtensionName}})); err != nil {
```

Comment:
Ideally, this would run under leader election. E.g.:

```go
// TODO(Wieneo): Remove this once a couple extension versions included the migration code
	// migration code: remove mutating webhook from cluster as it is not served by this controller anymore
	if err := mgr.Add(manager.RunnableFunc(func(ctx context.Context) error {
		if err := client.IgnoreNotFound(mgr.GetClient().Delete(ctx, &admissionregistrationv1.MutatingWebhookConfiguration{ObjectMeta: metav1.ObjectMeta{Name: ExtensionName}})); err != nil {
			return fmt.Errorf("could not delete mutatingwebhook %s: %s", ExtensionName, err)
		}
		return nil
	})); err != nil {
		return fmt.Errorf("could not add runnable to manager: %s", err)
	}
```